GitSearch/Networking/Encoding/URLEncoder.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  URLEncoder.swift
+//  GitSearch
+//
+//  Created by Vldmr on 8/26/19.
+//  Copyright © 2019 mudritskiy. All rights reserved.
+//
+
+import Foundation
+
+public struct URLEncoder {
+    
+    static func encodeParameters(for urlRequest: inout URLRequest, with parameters: HTTPParameters?) throws {
+        guard let url = urlRequest.url else { throw HTTPNetworkError.missingURL }
+        
+        if var urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: false), let parameters = parameters, !parameters.isEmpty {
```

Comment:
The whole purpose of `guard` in swift is that you do not have heavily nested code like this:
```
if firstCondition {
    if secondCondition {
        code
    }
}
```
vs
```
guard firstCondition else { return }
guard secondCondition else { return }

code
```
 Although for sure its possible to use `if` in swift (it obviously compiles and works), still it would be better to at least try language approach.

Reply:
Regarding usage patterns, `guard`s are used in cases when it does not make much sense to continue: preconditions must be fulfilled.

Reply:
it is hard for me, because eary I liked to use early exit opposite to `if`.
then was another paradigm - to use `if` rather then `guard` like construction.
and now to use guard again